app/indexers/concerns/newspaper_works/indexes_relationships.rb

```diff
@@ -72,6 +73,23 @@ def index_pages(article, solr_doc)
       end
     end
 
+    # index previous/next siblings info
+    #
+    # @param page [NewspaperPage]
+    # @param solr_doc [Hash] the hash of field data to be pushed to Solr
+    def index_siblings(page, solr_doc)
+      newspaper_issue = page.issue
+      return unless newspaper_issue.is_a?(NewspaperIssue)
+      page_ids = newspaper_issue.ordered_page_ids
+      return unless page_ids.length > 1
+      this_page_index = page_ids.index(page.id)
+      return unless this_page_index
+      unless this_page_index.zero?
+        solr_doc['is_following_page_of_ssi'] = page_ids[this_page_index - 1].presence
+      end
+      solr_doc['is_preceding_page_of_ssi'] = page_ids[this_page_index + 1].presence
```

Comment:
I wonder what happens here on last page, if `this_page_index == page_ids.length-1` — would you expect to see `NoMethodError: undefined method `presence' for nil:NilClass`?

Reply:
Can you clarify, I hope I understand correctly:

_Suppose I add a 3 page issue:_
- Save page 1 first, there is no page 2 yet, `is_preceding_page_of_ssi` is nil.
- Save page 2 next, page 1 has no `is_preceding_page_of_ssi` as a direct consequence of saving page 2, but there is a save hook as a consequence of saving the whole issue (in `CreateIssuePagesJob.perform`)?
- There probably is no graceful way to defer sibling indexing until the whole issue's ingest is done, at least without adding complexity differentiating between what calling the save (e.g. setting some kind of global state)?  Not knowing yet what the performance cost of this is, I think this (indexing each page twice) is completely acceptable for now.

Reply:
The `presence` method will return `nil` when called on `NilClass`, otherwise it returns the value. It's a pretty handy method in that respect. See definition [here](https://apidock.com/rails/Object/presence).

Your understanding of the process is correct, I think. The proper relationships won't be set as the pages are being added one by one because the next siblings don't exist yet. It's only the re-save at the end of `CreateIssuePagesJob.perform` where relationships have been established and therefore the indexing will be correct.

It would be nice if there was a way to not call the indexer on save, except in a specific context. I'm not sure how bad the performance hit is on this.

Reply:
Re: `presence` method: my concern was possibility of `page_ids[this_page_index + 1]` being nil for last page of sequence, which would, AFAICT, throw `NoMethodError`?

Reply:
If I'm understanding your comment, that wouldn't be a concern, because nil.presence does not return a NoMethodError, but just nil.
```
# rails console
2.4.1 :001 > nil.presence
 => nil 
2.4.1 :002 > nil.respond_to?(:presence)
 => true
```

Reply:
@ebenenglish ok, good to know (this must be a rails monkey patch of Ruby NilClass, as `nil.presence` does raise the exception in `irb`, but does not in `rails console` of a vanilla rails app).